src/main/java/com/es/phoneshop/web/ProductPageServlet.java

```diff
@@ -0,0 +1,37 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.model.product.ArrayListProductDao;
+import com.es.phoneshop.model.product.ProductDao;
+import com.es.phoneshop.model.product.ProductNotFoundException;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class ProductPageServlet extends HttpServlet {
+    private ProductDao productDao;
+
+    @Override
+    public void init() throws ServletException {
+        super.init();
+        productDao = ArrayListProductDao.getInstance();
+
+    }
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        try {
+            request.setAttribute("product", productDao.getProduct(parseProductId(request)));
+            request.getRequestDispatcher("/WEB-INF/pages/product.jsp").forward(request, response);
+        }
+        catch(ProductNotFoundException e) {
```

Comment:
Don't catch `ProductNotFoundException`. Map this exception to 404 HTTP code in web.xml and bind it to custom 404.jsp